### ISSUE TITLE
[Bug] Initialization of A_log and dt_bias in GatedDeltaNet

### ISSUE THREAD

**Issue by stiger1000** [fla-org/flame#42]
### Describe the Bug

I ran the training recipes to train a GatedDeltaNet, where I found that the initialization of A_log and dt_bias did not take effect, and both parameters were initialized to 0.
This problem only ocurs when NGPU > 1.

### Steps to Reproduce the Bug

run the following training recipes and monitor the value of A_log and dt_bias.

export NGPU=8
bash train.sh \
  --job.config_file flame/models/fla.toml \
  --job.dump_folder exp/gdn-340M \
  --model.config configs/gated_deltanet_340M.json \
  --model.tokenizer_path fla-hub/transformer-1.3B-100B \
  --optimizer.name AdamW \
  --optimizer.eps 1e-15 \
  --optimizer.lr 1e-3 \
  --lr_scheduler.warmup_steps 1024 \
  --lr_scheduler.lr_min 0.1 \
  --lr_scheduler.decay_type cosine \
  --training.batch_size 1 \
  --training.seq_len 65536 \
  --training.context_len 4096 \
  --training.varlen \
  --training.gradient_accumulation_steps 1 \
  --training.steps 20480 \
  --training.max_norm 1.0 \
  --training.skip_nan_inf \
  --training.dataset HuggingFaceFW/fineweb-edu \
  --training.dataset_name sample-100BT \
  --training.dataset_split train \
  --training.num_workers 32 \
  --training.prefetch_factor 2 \
  --training.seed 42 \
  --checkpoint.interval 2048 \
  --checkpoint.load_step -1 \
  --checkpoint.keep_latest_k 2 \
  --metrics.log_freq 1

### Expected Behavior

Both A_log and dt_bias are initialized to 0, which is not what we expect to see.

### Environment Information

1. Torch: 2.7.0+cu126
2. Triton: 3.3.0
